projects/igniteui-angular/src/lib/grid/grid.component.ts

```diff
@@ -4592,4 +4855,134 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
+            const lastIndex = this.rowList.length - 1;
+            const rowList = this.rowList.toArray();
+            if ((row.rowID === rowList[lastIndex].rowID ||
+                row.rowID === rowList[lastIndex - 1].rowID ||
+                row.rowID === rowList[lastIndex - 2].rowID)
+                // if row === rowList[0], then overlay should go to bottom, as otherwise it goes under the grid header
+                && row.rowID !== rowList[0].rowID) {
+                this.rowEditingOverlaySettings.positionStrategy.settings.verticalDirection = VerticalAlignment.Top;
+                this.rowEditingOverlaySettings.positionStrategy.settings.verticalStartPoint = VerticalAlignment.Top;
+            } else {
+                this.rowEditingOverlaySettings.positionStrategy.settings.verticalDirection = VerticalAlignment.Bottom;
+                this.rowEditingOverlaySettings.positionStrategy.settings.verticalStartPoint = VerticalAlignment.Bottom;
+            }
+
+            if (this.rowEditingOverlay.element.style.display === 'none') {
+                this.rowEditingOverlay.element.style.display = 'block';
+            }
+        }
+    }
+
+    public calculateRowChangesCount(rowID) {
+        const rowChanges = this.transactions.getAggregatedValue(rowID, false);
+        const rowChangedCount = rowChanges ? Object.keys(rowChanges).length : 0;
+        this.rowEditMessageValue = this.rowEditMessage.replace("{0}", rowChangedCount.toString());
+    }
+
+    /**
+     * @hidden
+     */
+    public hideRowEditingOverlay() {
+        this.rowEditingOverlay.element.style.display = 'none';
+    }
+
+    /**
+     * @hidden
+     */
+    public closeRowEditingOverlay(commit?: boolean) {
+        this.transactions.endPending(commit);
+        const row = this.gridAPI.get_row_inEditMode(this.id);
+        if (row) {
+            const value = this.transactions.getAggregatedValue(row.rowID);
+            this.onRowEditCancel.emit(value);
+        }
+        this.rowEditingOverlay.close();
+    }
+
+    /**
+     * @hidden
+     */
+    public endRowTransaction(commit?: boolean, closeOverlay?: boolean, row?: {rowID: any, rowIndex: number}) {
+        const rowInEdit = row ? row : this.gridAPI.get_row_inEditMode(this.id);
+        const value = this.transactions.getAggregatedValue(rowInEdit.rowID);
+        if (commit) {
+            this.onRowEditDone.emit(value);
+        } else {
+            this.onRowEditCancel.emit(value);
+        }
+        this.transactions.endPending(commit);
+        if (closeOverlay) {
+            this.closeRowEditingOverlay(closeOverlay);
+        }
+        this._pipeTrigger++;
+        this.cdr.detectChanges();
+    }
+
+    /**
+     * Finishes the row transactions on the current row.
+     * If `commit === true`, passes them from the pending state to the data (or transaction service)
+     * @param commit
+     */
+    public closeRowTransaction(commit?: boolean) {
```

Comment:
TBH, I'd skip that one for the first release, but while at it:
- Shouldn't it be just `endRowEdit(commit = false)`? Transactions are a separate *optional* module after all.
- If you do go with close/end I'd expect a start
- There's already API to enter cell edit that should trigger row edit, right? Assuming ending the cell edit will do the opposite, do you really need a row-specific method?